Algorithms/Ruby/rate_limiter.rb

```diff
@@ -0,0 +1,45 @@
+class MyRateLimitedService
```

Comment:
I would do this with a ten element array, and only letting a request go through (and adding it to the end) if the first element is more than ten seconds away.